### PR TITLE
UIE-204 Narrow Ajax Usage pt8

### PR DESCRIPTION
**Pull request by msilva-broad**
- shifting ajax usage to directly use ajax sub-areas for some terra-ui common components.
- improving mock types where possible.

### What
-

### Why
- improve code maintainability

### Testing strategy

- [ ]

### CODE REVIEW

**Review by trholdridge — APPROVED** [review]

File: src/components/WorkflowSelector.js [hunk 1]
```diff
@@ -45,7 +46,7 @@ const WorkflowSelector = ({
       style: { marginBottom: '1rem', height: 50, flex: 'none' },
       text: workflowName,
       onClick: () => {
-        Ajax().Metrics.captureEvent(Events.workspaceDataOpenWithWorkflow, extractWorkspaceDetails({ namespace, name }));
+        void Metrics().captureEvent(Events.workspaceDataOpenWithWorkflow, extractWorkspaceDetails({ namespace, name }));
```
Comment (s-rubenstein) [inline src/components/WorkflowSelector.js:49]: Why do you need the void here?

**Review by s-rubenstein — APPROVED** [review]
This looks good, one question for my understanding?

File: src/components/WorkflowSelector.js (hunk 1 above)
Comment (msilva-broad) [reply src/components/WorkflowSelector.js:49]: there's a lint warning on this, "fire and forget" async calls like this common Metrics call should be marked with void prefix to note that it is intentionally not being awaited on.

Comment (msilva-broad) [reply src/components/WorkflowSelector.js:49]: there's a useMetrics hook that avoids this awkwardness, but switching over to that is not part of this wave of cleanup to keep things narrow.

**Pull request merged**